presto-spi/src/main/java/com/facebook/presto/spi/InMemoryRecordSet.java

```diff
@@ -118,7 +120,13 @@ public long getLong(int field)
         {
             checkState(record != null, "no current record");
             requireNonNull(record.get(field), "value is null");
-            return ((Number) record.get(field)).longValue();
+            Object value = record.get(field);
```

Comment:
Perhaps add a preconditions check to ensure we don't return just the unscaled value of a `LongDecimalType`?

Reply:
Sure, good suggestion. Fixed!